File: app/src/main/java/com/turastory/kotlindemo/todo/app/DeviceConfiguration.java
package com.turastory.kotlindemo.todo.app;

import android.content.Context;
import android.content.res.Configuration;

/**
 * Util class - Extension functions
 */
public class DeviceConfiguration {
    public static boolean isTablet(Context context) {
        boolean xlarge = screenConfig(context) == Configuration.SCREENLAYOUT_SIZE_XLARGE;
        boolean large = screenConfig(context) == Configuration.SCREENLAYOUT_SIZE_LARGE;
        return xlarge || large;
    }
    
    private static int screenConfig(Context context) {
        return config(context).screenLayout & Configuration.SCREENLAYOUT_SIZE_MASK;
    }
    
    private static Configuration config(Context context) {
        return context.getApplicationContext().getResources().getConfiguration();
    }
}

File: app/src/main/java/com/turastory/kotlindemo/kotlin/app/DeviceConfiguration.kt
@file:JvmName("DeviceConfiguration")

package com.turastory.kotlindemo.kotlin.app

import android.content.Context
import android.content.res.Configuration

/**
 * Util class - Extension functions
 */
fun Context.isTablet(): Boolean {
    val xlarge = screenConfig() == Configuration.SCREENLAYOUT_SIZE_XLARGE
    val large = screenConfig() == Configuration.SCREENLAYOUT_SIZE_LARGE
    return xlarge || large
}

private fun Context.screenConfig(): Int {
    return this.config().screenLayout and Configuration.SCREENLAYOUT_SIZE_MASK
}

private fun Context.config(): Configuration {
    return this.applicationContext.resources.configuration
}